pulp_ansible/app/migrations/0036_update_repository_content.py

```diff
@@ -0,0 +1,76 @@
+# Generated by Django 2.2.24 on 2021-07-07 17:46
+
+from collections import defaultdict
+from django.db import migrations
+
+
+def create_repository_content_for_deprecation(apps, schema_editor):
+    """ Create RepositoryContent for AnsibleCollectionDeprecated."""
+    AnsibleCollectionDeprecated = apps.get_model('ansible', 'AnsibleCollectionDeprecated')
+    RepositoryContent = apps.get_model('core', 'RepositoryContent')
+    RepositoryVersionContentDetails = apps.get_model('core', 'RepositoryVersionContentDetails')
+    db_alias = schema_editor.connection.alias
+    repo_content = []
+    counts_list = []
+    counter = defaultdict(int)
+    for deprecated_collection in AnsibleCollectionDeprecated.objects.using(db_alias).all():
+        counter[str(deprecated_collection.version_added_id)] += 1
+        repo_content.append(RepositoryContent(
+            content=deprecated_collection,
+            repository_id=deprecated_collection.repository_id,
+            version_added_id=deprecated_collection.version_added_id,
+        ))
+
+    for repo_version_id, total in counter.items():
+        count_obj = RepositoryVersionContentDetails(
+            content_type=deprecated_collection.pulp_type,
+            repository_version_id=repo_version_id,
+            count=total,
```

Comment:
I don't know that this is correct. Say I have 2 deprecated collections in repo version 2 but version 1 has 1 of them too. Then this number should be 1 and not 2. 

Also, I think you'll need to calculate the deprecations removed as well.

Reply:
Can't we just call repo_version._compute_counts() ?

Reply:
ORM methods don't work here

Reply:
Actually this is even more of a huge pain because I think we'll need to delete any existing RepositoryVersionContentDetails first and then totally recompute them:

https://github.com/pulp/pulpcore/blob/1d76e8a316a5fc92cc3397f01329be10fe68a7a5/pulpcore/app/models/repository.py#L926

Maybe we should just not compute `RepositoryVersionContentDetails` in our migration and add a note how users can recompute these themselves.

Reply:
Oh I see. :( I agree

Reply:
I'm ok to leave these counts as is, and if it's a problem folks can file a new issue about them being incorrect.

Reply:
To be clear, you're suggesting we don't update RepositoryVersionContentDetails (and thus remove this part of the migration)? If so, +1 from me.

Reply:
Actually, RepositoryVersionContentDetails uncovered a big issue,

> I don't know that this is correct. Say I have 2 deprecated collections in repo version 2 but version 1 has 1 of them too. Then this number should be 1 and not 2.

> Also, I think you'll need to calculate the deprecations removed as well.

I have to check if deprecations were removed at some point, right now I only do `version_added`, I need to cover `version_removed` as well

```python
repo_content.append(RepositoryContent(
            content=deprecated_collection,
            repository_id=deprecated_collection.repository_id,
            version_added_id=deprecated_collection.version_added_id,
        ))
```

Reply:
Yes, I think you may need to change how you calculate version_added as well. If a deprecation is in versions 1, 2, 3 then there should only be one RepositoryContent with version_added=1.

Reply:
can I have another round of review?
I've improved the migration, but I'm not sure if it is covering version_added/version_removed well